src/Tools/dotnet-monitor/EndpointInfo/ServerEndpointInfoSource.cs

```diff
@@ -84,21 +84,25 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             if (_portOptions.ConnectionMode == DiagnosticPortConnectionMode.Listen)
             {
-                if (_portOptions.GetDeleteEndpointOnStartup() &&
-                   !RuntimeInformation.IsOSPlatform(OSPlatform.Windows) &&
-                    File.Exists(_portOptions.EndpointName))
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                 {
-                    // In some circumstances stale files from previous instances of dotnet-monitor cause
-                    // the new instance to fail binding. We need to delete the file in this situation.
-                    try
-                    {
-                        _logger.DiagnosticPortDeleteAttempt(_portOptions.EndpointName);
-                        File.Delete(_portOptions.EndpointName);
-                    }
-                    catch (Exception ex)
+                    if (_portOptions.GetDeleteEndpointOnStartup() &&
+                        File.Exists(_portOptions.EndpointName))
                     {
-                        _logger.DiagnosticPortDeleteFailed(_portOptions.EndpointName, ex);
+                        // In some circumstances stale files from previous instances of dotnet-monitor cause
+                        // the new instance to fail binding. We need to delete the file in this situation.
+                        try
+                        {
+                            _logger.DiagnosticPortDeleteAttempt(_portOptions.EndpointName);
+                            File.Delete(_portOptions.EndpointName);
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.DiagnosticPortDeleteFailed(_portOptions.EndpointName, ex);
+                        }
                     }
+
+                    Directory.CreateDirectory(Path.GetDirectoryName(_portOptions.EndpointName));
```

Comment:
Is this method safe if the directory already exists?

Reply:
Yes